A41-xds-rbac.md

```diff
@@ -0,0 +1,230 @@
+request to be used by RBAC and the application itself.
+
+In RBAC `metadata` refers to the Envoy metadata which has no relation to gRPC
+metadata. Envoy metadata is generic state shared between filters, which has no
+gRPC equivalent. RBAC implementations in gRPC will treat Envoy metadata as an
+empty map. Since `ValueMatcher` can only match if a value is present (even
+`NullMatch`), the `metadata` matcher is guaranteed not to match.
+
+`requested_server_name` can match if the matcher accepts empty string.
+
+The following fields of `Principal` may not be obvious how they map to gRPC:
+
+| Principal Field  | gRPC Equivalent |
+| ---------------- | --------------- |
+| authenticated.principal_name | The URI/DNS SAN or Subject; same as Envoy |
+| source_ip        | Peer address for this connection |
+| direct_remote_ip | Peer address for this connection |
+| remote_ip        | Peer address for this connection |
+| header           | Same as in Permission |
+| url_path         | Same as in Permission |
+| metadata         | Same as in Permission |
+
+The `authenticated.principal_name` will use the same definition as its
+`rbac.proto` comment, although it checks multiple values which isn't clear from
+the comment. If `principal_name` is unset, then `Authenticated` is said to match
+if the connection uses TLS; a client certificate is not necessary. Other
+values being checked are derived from the client's certificate. The process is:
+
+1. Check if any SubjectAltName entry with URI type (type 6) matches. If any
+   entry matches, the `principal_name` is said to match
+2. If there is no SAN with URI type, check if any SAN entry with DNS type (type
+   2) matches. If any entry matches, the `principal_name` is said to match
+3. If there are no SAN with URI or DNS types, check if the Subject's
+   distinguished name formatted as an RFC 2253 Name matches. If it matches, the
+   `principal_name` is said to match
+4. If there is no client certificate (thus no SAN nor Subject), check if `""`
+   (empty string) matches. If it matches, the `principal_name` is said to match
+
+`source_ip` is the same as `direct_remote_ip` only as long as
+[envoy.extensions.filters.listener.proxy_protocol.v3.ProxyProtocol][] is
+unsupported. If a future gRFC adds ProxyProtocol support it must also update
+`source_ip` and `remote_ip` handling in RBAC.
+
+`remote_ip` is the same as `direct_remote_ip` only as long as ProxyProtocol and
+`xff_num_trusted_hops` are unsupported.
+
+[RBAC policy]: https://github.com/envoyproxy/envoy/blob/10c17a7cd90b013c38dfbfbf715d3c24fdd0477c/api/envoy/config/rbac/v3/rbac.proto
+[RBAC network filter]: https://github.com/envoyproxy/envoy/blob/main/api/envoy/extensions/filters/network/rbac/v3/rbac.proto
+[envoy.extensions.filters.listener.proxy_protocol.v3.ProxyProtocol]: https://github.com/envoyproxy/envoy/blob/main/api/envoy/extensions/filters/listener/proxy_protocol/v3/proxy_protocol.proto
+
+### Temporary environment variable protection
```

Comment:
Another general question, does the gRPC RBAC emit the same stats for the evaluation result (e.g. allowed, denied number of requests), those are useful stats for quickly checking the status of the RBAC filter in Envoy.

Reply:
There are no stats as part of this. There have been some other discussions about stats (outside of RBAC) as stats are getting fleshed out. But as of yet, filters aren't contributing to stats. gRPC stats will not be the same as Envoy stats, but we'll obviously be aware of the Envoy stats and want similar stats for the really helpful cases.